### PR TITLE
Solution

### PR DESCRIPTION
**Pull request by ia-lerimund**

### CODE REVIEW

File: src/transformStateWithClones.js [hunk 1]
```diff
@@ -8,6 +8,27 @@
...
+      for (const key of actions[prop].keysToRemove) {
+        delete newObj[key];
+      }
+      result.push({ ...newObj });
+    } else if ((actions[prop].type === 'clear')) {
+      for (const key in newObj) {
+        delete newObj[key];
+      }
+      result.push({});
+    }
```
Comment (Bohdan-Maksymiuk) [inline src/transformStateWithClones.js:28]: avoid such nesting `else if` conditions
This code is hard to read and maintain
use `switch` instead

File: src/transformStateWithClones.js [hunk 2]
```diff
@@ -8,6 +8,27 @@
  */
 function transformStateWithClones(state, actions) {
   // write code here
+  const newObj = Object.assign({}, state);
+  const result = [];
+
+  for (let prop = 0; prop < actions.length; prop++) {
```
Comment (Bohdan-Maksymiuk) [inline src/transformStateWithClones.js:14]: use `for of` loop
you don't need indexes here

**Review by Bohdan-Maksymiuk — CHANGES_REQUESTED** [review]

File: src/transformStateWithClones.js (hunk 1 above)
Comment (ia-lerimund) [reply src/transformStateWithClones.js]: Fixed it, thanks for the advice!

File: src/transformStateWithClones.js [hunk 3]
```diff
@@ -8,6 +8,30 @@
...
+          delete newObj[key];
+        }
+        result.push({ ...newObj });
+        break;
+      case 'clear':
+        for (const key in newObj) {
+          delete newObj[key];
+        }
+        result.push({});
+    }
```
Comment (witflash) [inline src/transformStateWithClones.js:31]: You need to add `default` statement for `switch / case` block. 
You can handle errors there, for example

**Review by witflash — CHANGES_REQUESTED** [review]
Good job!
But check my last comment, please.

**Pull request closed without merging**